src/components/v2/V2Create/tabs/TokenTabContent.tsx

```diff
@@ -1,39 +1,235 @@
+
+  const {
+    fundingCycleMetadata,
+    fundingCycleData,
+    fundAccessConstraints,
+    reserveTokenSplits: reduxReserveTokenSplits,
+  } = useAppSelector(state => state.editingV2Project)
+
+  const reduxDiscountRate = fundingCycleData?.discountRate
+  const reduxReservedRate = fundingCycleMetadata?.reservedRate
+  const reduxRedemptionRate = fundingCycleMetadata?.redemptionRate
+
+  const [reserveTokenSplits, setReserveTokenSplits] = useState<TicketMod[]>(
+    reduxReserveTokenSplits ?? [],
+  )
+
+  const [discountRateDisabled, setDiscountRateDisabled] = useState<boolean>(
+    reduxDiscountRate === undefined || reduxDiscountRate === '0',
+  )
+
+  const [reservedRateDisabled, setReservedRateDisabled] = useState<boolean>(
+    reduxReservedRate === undefined || reduxReservedRate === '0',
+  )
+
+  const [redemptionRateDisabled, setRedemptionRateDisabled] = useState<boolean>(
+    reduxRedemptionRate === undefined || reduxRedemptionRate === '100',
+  )
 
-  const onTokenFormSaved = useCallback(
-    (fields: TokenFormFields) => {
-      dispatch(editingV2ProjectActions.setDiscountRate(fields.discountRate))
-      dispatch(editingV2ProjectActions.setReservedRate(fields.reservedRate))
-    },
-    [dispatch],
+  // Using a state here because relying on the form does not
+  // pass through updated reservedRate to ProjectTicketMods
+  const [reservedRate, setReservedRate] = useState<number | undefined>(
+    parseFloat(reduxReservedRate),
   )
 
+  const dispatch = useAppDispatch()
+
+  const onTokenFormSaved = useCallback(() => {
+    const fields = tokenForm.getFieldsValue(true)
+    dispatch(editingV2ProjectActions.setDiscountRate(fields.discountRate))
+    dispatch(editingV2ProjectActions.setReservedRate(fields.reservedRate))
+    dispatch(editingV2ProjectActions.setRedemptionRate(fields.redemptionRate))
+    dispatch(editingV2ProjectActions.setReserveTokenSplits(reserveTokenSplits))
+  }, [dispatch, tokenForm, reserveTokenSplits])
+
   const resetTokenForm = useCallback(() => {
     tokenForm.setFieldsValue({
-      discountRate: fundingCycleData?.discountRate ?? '0',
+      discountRate: reduxDiscountRate ?? '0',
+      reservedRate: reduxReservedRate ?? '0',
+      redemptionRate: reduxRedemptionRate ?? '100',
     })
-  }, [fundingCycleData?.discountRate, tokenForm])
+    setReserveTokenSplits(reserveTokenSplits)
+  }, [
+    reduxDiscountRate,
+    reduxReservedRate,
+    reduxRedemptionRate,
+    reserveTokenSplits,
+    tokenForm,
+  ])
+
+  const disableTextStyle: CSSProperties = {
+    color: colors.text.primary,
+    fontStyle: 'italic',
+    fontWeight: 500,
+    marginBottom: 10,
+  }
 
   // initially fill form with any existing redux state
   useEffect(() => {
     resetTokenForm()
   }, [resetTokenForm])
 
+  // Assume the first item is the one of interest.
+  const fundAccessConstraint = fundAccessConstraints[0] as
+    | SerializedV2FundAccessConstraint
+    | undefined
+
+  const _hasFundingTarget = Boolean(
```

Comment:
**suggestion** can we avoid underscore in this name?

Reply:
Removed these and used the new functions in `utils/fundingCycleV2`

Reply:
respecc